src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -788,78 +795,187 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 	}
 
 	private async _executeInTerminal(task: CustomTask | ContributedTask, trigger: string, resolver: VariableResolver, workspaceFolder: IWorkspaceFolder | undefined): Promise<ITaskSummary> {
+		let result: ITaskExecutionResult;
+		if (task.configurationProperties.isBackground) {
+			result = await this._executeBackgroundTask(task, resolver, workspaceFolder);
+		} else {
```

Comment:
Hard to review this, it got a bit refactor too?

Reply:
yes pulled it out

Reply:
can do more refactoring/add tests in a diff't PR as this one is already large